tests/test_protocol.py

```diff
@@ -1,18 +1,101 @@
+            [0,	0,	0,	1]
+        ]
+    )
+
+@pytest.fixture()
+def example_volume() -> xa.Dataset:
+    #TODO the following is in case we want to test a nifti input volume
+    # loading a nifti file and then construction a xarray dataset from it
+    # with open(Path(__file__).parent/"resources/example_db/subjects/example_subject/volumes/example_volume/mni.json") as f:
+    #     nib_volume_metadata = json.load(f)
+    # nib_volume = Nifti1Image.load(Path(__file__).parent/"resources/example_db/subjects/example_subject/volumes/example_volume"/nib_volume_metadata['data_filename'])
+    # l_min, p_min, s_min = nib_volume.affine[:3, -1]
+    # vol_shape = nib_volume.shape
+    # l_max = affines.apply_affine(nib_volume.affine, [vol_shape[0]-1, 0, 0])[0]
+    # p_max = affines.apply_affine(nib_volume.affine, [0, vol_shape[1]-1, 0])[1]
+    # s_max = affines.apply_affine(nib_volume.affine, [0, 0, vol_shape[2]-1])[2]
+    # volume = xa.Dataset(
+    #     {
+    #         'data': xa.DataArray(data=nib_volume.get_fdata(), dims=["L", "P", "S"])
+    #     },
+    #     coords={
+    #         'L': xa.DataArray(dims=["L"], data=np.linspace(l_min, l_max, vol_shape[0]), attrs={'units': "mm"}),
+    #         'P': xa.DataArray(dims=["P"], data=np.linspace(p_min, p_max, vol_shape[1]), attrs={'units': "mm"}),
+    #         'S': xa.DataArray(dims=["S"], data=np.linspace(s_min, s_max, vol_shape[2]), attrs={'units': "mm"})
+    #     },
+    #     attrs={
+    #         'units': "",
+    #         'name': "Test Volume",
+    #         'id': "volume_000",
+    #         'affine': nib_volume.affine,
+    #         'affine_units': "mm"
+    #     }
+    # )
+    return None
+
 @pytest.mark.parametrize("compact_representation", [True, False])
 def test_serialize_deserialize_protocol(example_protocol : Protocol, compact_representation: bool):
     assert example_protocol.from_json(example_protocol.to_json(compact_representation)) == example_protocol
 
 def test_default_protocol():
     """Ensure it is possible to construct a default protocol"""
     Protocol()
+
+def test_calc_solution(
+        example_protocol: Protocol,
+        example_transducer: Transducer,
+        example_session: Session,
+        example_transducer_transform: np.ndarray,
+        example_volume: xa.Dataset
+    ):
+    """Make sure a solution can be calculated."""
+    import logging
+    from copy import deepcopy
+
+    from openlifu.util.units import convert_transform
+
+    logging.disable(logging.CRITICAL)
+    target = deepcopy(example_session.targets[0])
+    target.rescale("m")
+    transducer_transform = convert_transform(example_transducer_transform, units="mm", tgt_units="m")
+    target.transform(np.linalg.inv(transducer_transform))
+    target.dims = ("lat", "ele", "ax")
+
+    example_protocol.calc_solution(
+        target,
+        example_transducer,
+        transducer_transform,
+        volume=example_volume,
+        session=example_session,
+        simulate=False,
```

Comment:
I tried different values for `dt` and `t_end` but that does not have impact so I am not sure what I am doing wrong here...

Reply:
How about this. Let's remove `test_calc_solution` and we will plan to add testing later.

I don't want to keep `test_calc_solution` because it will provide fake testing coverage when the only thing being tested is that the code runs without raising errors.

If we remove it then we will see that it part of the untested code and remember to test it later.

Reply:
should we do the same for solution analysis ?